Zend/zend_vm_def.h

```diff
@@ -10062,6 +10062,19 @@ ZEND_VM_HOT_TYPE_SPEC_HANDLER(ZEND_IS_NOT_EQUAL|ZEND_IS_NOT_IDENTICAL, (op1_info
 	ZEND_VM_SMART_BRANCH(result, 0);
 }
 
+ZEND_VM_TYPE_SPEC_HANDLER(ZEND_IS_IDENTICAL, op->op2_type == IS_CONST && (Z_TYPE_P(RT_CONSTANT(op, op->op2)) == IS_ARRAY && zend_hash_num_elements(Z_ARR_P(RT_CONSTANT(op, op->op2))) == 0), ZEND_IS_IDENTICAL_EMPTY_ARRAY, TMPVARCV, CONST, SPEC(SMART_BRANCH,COMMUTATIVE))
```

Comment:
Why do I manually need to check `op->op2_type == IS_CONST` here? Shouldn't that be the job of the `CONST` in `ZEND_IS_IDENTICAL_EMPTY_ARRAY, TMPVARCV, CONST`?

Reply:
More generally speaking, I'm not at all sure if the “flags” I specified here are correct, since I clearly don't understand the implications. So please carefully check this :smile:

Reply:
You can look at the generated `zend_vm_set_opcode_handler_ex()`. It assumes op types are correct. If it works only for some op types, checking the op type should be the right approach. The signature looks correct to me.